docs/quickstart/basics.rst

```diff
@@ -60,38 +78,49 @@ Request Body Validation
 
 .. code-block:: python
 
-   from flask_rebar import RequestSchema
-
-
-   class CreateTodoSchema(RequestSchema):
-       description = fields.String(required=True)
-
-
    @registry.handles(
       rule='/todos',
       method='POST',
-      request_body_schema=CreateTodoSchema(),
+      request_body_schema=TodoSchema(strict=True),
+      response_body_schema={201: TodoSchema()}
    )
    def create_todo():
-       body = rebar.validated_body
-       description = body['description']
-       . . .
+       # If we got here, it means we have valid Todo data in the request body.
+       # Thanks to specifying the request_body_schema above, Rebar takes care
+       # of sending nice 400 responses (with human- and machine-friendly bodies)
+       # in response to invalid request data for us.
+       description = rebar.validated_body['description']
+       new_todo = _create_todo(description)  # Create a Todo in our db and return it.
+       # We'll want to return a 201 Created response with a Location header, so use
+       # flask.url_for to get the URL of the new Todo:
+       new_todo_url = flask.url_for(get_todo, id=new_todo.id)
+       response_data = {"id": new_todo.id, "description": new_todo.description}
+       return (response_data, 201, {"Location": new_todo_url})
 
 
-``RequestSchema`` is an extension of ``marshmallow.Schema`` that throws an internal server error if an object is missing a required field. It's usage is optional - a normal Marshmallow schema will also work.
-
 This request schema is passed to ``request_body_schema``, and the handler will now call `marshmallow.Schema.load <http://marshmallow.readthedocs.io/en/latest/api_reference.html#marshmallow.Schema.load>`_ on the request body decoded as JSON. A 400 error with a descriptive error will be returned if validation fails.
```

Comment:
RequestSchema is still important for marshmallow 2 if I am not mistaken, otherwise additional fields are not rejected. What I think we should do in V2 is a single Schema class that includes the RequestSchema and ResponseSchema, but I will open an issue for that.

Reply:
See https://github.com/plangrid/flask-rebar/pull/152#discussion_r373322964